src/IECoreMaya/SceneShapeInterface.cpp

```diff
@@ -824,6 +825,34 @@ MStatus SceneShapeInterface::compute( const MPlug &plug, MDataBlock &dataBlock )
 					{
 						MFnNurbsCurveData fnData;
 						data = fnData.create();
+
+						// If "index" convert paramter is specified, it determines which one of the curves to output in the Curves primitive.
+						std::map<std::string, std::string> convertParams;
+						bool isParamsRead = readConvertParams( convertParams, index );
```

Comment:
My last comment would be better placed here. Doesn't all this special casing go away if we just use `ParameterParser` directly?